### PR TITLE
Introduce some overloads IEquatable

### PR DESCRIPTION
**Pull request by ndrwrbgs**
The library exposes `IComparer` operations, but where it's implemented, `IEquatable` can be more efficient. This review provides the option for developers to utilize that interface if desired/type-inferrence applicable

### CODE REVIEW

File: src/projects/EnsureThat/ComparableArg.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ namespace EnsureThat
     public class ComparableArg
     {
         [DebuggerStepThrough]
+        [Obsolete("Prefer EquatableArg.IsNot where possible")]
```
Comment (danielwertheim) [inline src/projects/EnsureThat/ComparableArg.cs]: "Prefer EquatableArg.Is where possible"

**Comment by ndrwrbgs** [comment]
> This will introduce a breaking change since e.g. existing EnsureArg.Is now uses IEquatable<T> instead of IComparable<T>

Could introduce the Equatable as Equals() instead of Is(). Or could leave EnsureArg itself as IComparable, but then `EnsureArg` becomes a second-class citizen to `Ensure.That` and `Ensure.Equatable`

**Comment by ndrwrbgs** [comment]
I modified it to Equals to prevent the breaking change and merged with master.
Also, we should conflate the implementations. Without inlining, it's more efficient to duplicate the code (e.g. EnsureArg.* delegates to Ensure.Type.* but Ensure*Extensions duplicates the logic), but with AggressiveInlining it should be equivalent perf and allow us to have only 1 implementation of each method.

**Comment by danielwertheim** [comment]
@ndrwrbgs  Could we close all opened PRs or is there anything we feel like we really, really should fix?

**Comment by ndrwrbgs** [comment]
I mean I feel like everything should be fixed :P

I'll take a pass over the open ones but I think last I checked they were
hoping for some love from you to add features? :)

**Comment by ndrwrbgs** [comment]
@danielwertheim is there anything I can do to make the feature additions less work for you?

**Comment by danielwertheim** [comment]
@ndrwrbgs For me it's more like... All open PRs (not just this one) are getting somewhat old now so I don't really remember ;-) And to get them in now they need to be updated with the latest changes in master.

**Comment by ndrwrbgs** [comment]
Ah, okay, I will revisit them! Let's say next action on all of them is one me :) Thanks for the engagement @danielwertheim!

**Comment by danielwertheim** [comment]
I can dig in as well, I just wanted to ensure that the PRs doesn't contain anything we can discard... as in do we feel it's "worth" putting in some effort as they have been lying here for a while so to speak.

**Comment by ndrwrbgs** [comment]
Let me revisit to determine that, I'd belay any effort from your side until I do check that :)

**Comment by danielwertheim** [comment]
Seen as "expired". Closing.

**Comment by ndrwrbgs** [comment]
Hehe that's one way to make me revisit things :-D

**Pull request closed without merging**